### PR TITLE
feat: add framework, filename, and computed_name to Test model

### PR DESCRIPTION
**Pull request by joseph-sentry**

### CODE REVIEW

File: shared/django_apps/reports/migrations/0027_test_framework.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.2.15 on 2024-09-23 18:23
```
Comment (Swatinem) [inline shared/django_apps/reports/migrations/0027_test_framework.py:1]: this file conflicts with the other migration file

File: shared/django_apps/reports/migrations/0027_test_computed_name_test_filename_test_framework.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        ),
+        migrations.AddField(
+            model_name="test",
+            name="framework",
+            field=models.CharField(
+                choices=[
+                    ("pytest", "Pytest"),
+                    ("jest", "Jest"),
+                    ("vitest", "Vitest"),
+                    ("phpunit", "Phpunit"),
```
Comment (ajay-sentry) [inline shared/django_apps/reports/migrations/0027_test_computed_name_test_filename_test_framework.py:47]: PHPUnit vs. Phpunit assuming these are the legitimate capitalizations

File: shared/django_apps/reports/migrations/0027_test_computed_name_test_filename_test_framework.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            model_name="test",
+            name="filename",
+            field=models.TextField(null=True),
+        ),
+        migrations.AddField(
+            model_name="test",
+            name="framework",
+            field=models.CharField(
+                choices=[
+                    ("pytest", "Pytest"),
```
Comment (ajay-sentry) [inline shared/django_apps/reports/migrations/0027_test_computed_name_test_filename_test_framework.py:44]: Pytest vs pytest

**Review by ajay-sentry — APPROVED** [review]
lgtm!

**Pull request merged**